Testscripts/Linux/SR-IOV_Interrupts_Change.sh

```diff
@@ -0,0 +1,151 @@
+        fi
+    done < "/proc/interrupts"
+
+    if [ $(($vf_count*9)) -gt $pci_processing_interrupts_count ]; then
+        LogErr "The number of PCIe MSI devices processing interrupts is smaller than expected!"
+        SetTestStateFailed
+        exit 0
+    else
+        LogMsg "The number of MSI PCIe devices processing interrupts is: $pci_processing_interrupts_count"
+    fi
+
+    # $1 -eq 1 means it will construct the initial array of the interrupts/PCIe
+    # $1 -eq 2 means it will construct the final array of the interrupts/PCIe
+    if [ "$1" -eq "1" ]; then
+        initial_pci_interrupts=( "${pci_interrupt_values[@]}" )
+    else
+        final_pci_interrupts=( "${pci_interrupt_values[@]}" )
+    fi
+}
+
+function get_interrupts_per_vcpu() {
+    cpu_processing_interrupts_count=0
+    cpu_array=()
+    for (( vcpu_line=1; vcpu_line<=$(nproc); vcpu_line++ ))
+    do
+        line=$(cat /proc/interrupts | grep -i MSI | awk '{$1=""} NF{NF-=5}1' | ruby -e'puts readlines.map(&:split).transpose.map{|x|x*" "}' | head -"$vcpu_line" | tail -1)
+        vcpu_number=$(($vcpu_line-1))
+        interrupts_sum=$(echo $line | awk '{sum=0; for(i=1; i<=NF; i++) sum += $i; print sum}')
+        cpu_array[$vcpu_number]+=$interrupts_sum
+        LogMsg "vCPU $vcpu_number is processing $interrupts_sum interrupts"
+        if [ "$interrupts_sum" -gt "0" ]; then
+            ((cpu_processing_interrupts_count++))
+        fi
+    done
+
+    # $1 -eq 1 means it will construct the initial array of the interrupts/vCPU
+    # $1 -eq 2 means it will construct the final array of the interrupts/vCPU
+    if [ "$1" -eq "1" ]; then
+        initial_cpu_interrupts=( "${cpu_array[@]}" )
+    else
+        final_cpu_interrupts=( "${cpu_array[@]}" )
+    fi
+}
+
+if [ ! -e sriov_constants.sh ]; then
+    cp /${remote_user}/sriov_constants.sh .
+fi
+export PATH="/usr/local/bin:${PATH}"
+
+# Source SR-IOV_Utils.sh
+. SR-IOV-Utils.sh || {
+    echo "ERROR: unable to source SR-IOV_Utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+UtilsInit
+
+# Install required package
+update_repos
+install_package ruby
+
+# Check if the VF count inside the VM is the same as the expected count
+vf_count=$(find /sys/devices -name net -a -ipath '*vmbus*' | grep pci | wc -l)
+if [ "$vf_count" -ne "$NIC_COUNT" ]; then
+    LogErr "Expected VF count: $NIC_COUNT. Actual VF count: $vf_count"
+    SetTestStateFailed
+    exit 0
+fi
+UpdateSummary "Expected VF count: $NIC_COUNT. Actual VF count: $vf_count"
+
+# Get initial values of PCIe MSI devices processing interrupts and
+# values for each vCPU
+get_interrupts_per_pci "1"
+# Compute an array with interrupts/vCPU before the traffic starts
+get_interrupts_per_vcpu "1"
+
+# Start iPerf3 to force the interrupt count to increase
+LogMsg "------------ Running iPerf3 for 10 minutes ------------"
+ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$VF_IP2" "nohup iperf3 -s > /dev/null 2>&1 &"
```

Comment:
Should we check if iperf3 exists or not?

Reply:
It is always installed here: https://github.com/LIS/LISAv2/blob/master/Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1#L112